website/members/views.py

```diff
@@ -321,6 +321,162 @@ def get(self, request):
+                sent_error += 1
+                logger.error(
+                    "Problems sending email [%s] to member %s: %r", errors_code, member, err)
+            else:
+                sent_ok += 1
+            finally:
+                if missing_info['missing_signed_letter']:
+                    try:
+                        os.unlink(letter_filepath)
+                    except Exception as exc:
+                        logger.warning("Couldn't remove letter file %r: %r", letter_filepath, exc)
+
+        deltat = time.time() - tini
+        context = {
+            'sent_ok': sent_ok,
+            'sent_error': sent_error,
+            'errors_code': errors_code,
+            'deltamsec': int(deltat * 1000),
+        }
+        return render(request, 'members/mail_sent.html', context)
+
+    def _analyze_member(self, member):
+        """Analyze and indicate in which categories the member is missing somethhing."""
+        cat_student = Category.objects.get(name=Category.STUDENT)
+        cat_collab = Category.objects.get(name=Category.COLLABORATOR)
+        # simple flags with "Not Applicable" situation
+        missing_student_certif = (
+            member.category == cat_student and member.has_student_certificate)
+
+        missing_collab_accept = (
+            member.category == cat_collab and member.has_collaborator_acceptance)
+
+        # info from Person
+        missing_nickname = member.person.nickname != ""
+        # picture is complicated, bool() is used to check if the Image field has an associated
+        # filename, and False itself is used as the "dont want a picture!" flag
+        picture = member.person.picture is not None
+
+        # info from Member itself
+        missing_payment = member.first_payment_month is not None and member.category.fee != 0
+        missing_signed_letter = member.has_subscription_letter
+        missing = missing_signed_letter and missing_nickname and picture
+
+        return {
+            'missing': missing,
+            'missing_student_certif': missing_student_certif,
+            'missing_payment': missing_payment,
+            'missing_collab_accept': missing_collab_accept,
+        }
+
+    def get(self, request):
+        not_yet_members = Member.objects.filter(legal_id=None).order_by('created').all()
+
+        completes = []
+        k = 0
+        for member in not_yet_members:
+            info = self._analyze_member(member)
+
+            # convert info to proper strings to show
+            #print(k)
+            if info['missing'] and info['missing_student_certif']:
+                info[k] = "COMPLETO"
+            else:
+                if info['missing'] and info['missing_payment']:
+                    info[k] = "COMPLETO"
+                else:
+                    if info['missing'] and info['missing_collab_accept']:
+                        info[k] = "COMPLETO"
+                    else:
+                        info[k] = "INCOMPLETO"
+
```

Comment:
All this is superfluous. `_analyze_member` should be called something like `_is_member_complete` and return a bool.